apiserver/metricsender/metricsender_test.go

```diff
@@ -93,6 +93,38 @@ func (s *MetricSenderSuite) TestSendMetrics(c *gc.C) {
 	c.Assert(sent2.Sent(), jc.IsTrue)
 }
 
+// TestSendMetricsAbort creates 7 unsent metrics and
+// checks that the sending stops when no more batches are ack'ed.
+func (s *MetricSenderSuite) TestSendMetricsAbort(c *gc.C) {
+	sender := &testing.MockSender{}
+	now := time.Now()
+	metrics := make([]*state.MetricBatch, 7)
+	for i := 0; i < 7; i++ {
+		metrics[i] = s.Factory.MakeMetric(c, &factory.MetricParams{Unit: s.credUnit, Time: &now})
+	}
+
+	sender.IgnoreBatches(metrics[0:2]...)
+
+	// Send 4 batches per POST.
+	err := metricsender.SendMetrics(s.State, sender, s.clock, 4, true)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(sender.Data, gc.HasLen, 4)
```

Comment:
Why did we call sender.Send(...) 4 times instead of three? This is how I think it happened, am I correct?

Call#1: 1st 4 metrics sent, 2 acknowledged.
Call#2: same 2 unack'd metrics plus 2 more, 2 acknowledged.
Call#3: same 2 unack'd metrics plus 1 more, 1 acknowledged.
Call#4: same 2 unack'd metrics, 0 acknowledged, exiting the sender loop.